addons/mail/static/src/core/common/store_service.js

```diff
@@ -312,6 +316,197 @@ export class Store extends BaseStore {
 
     /** Provides an override point for when the store service has started. */
     onStarted() {}
+
+    /**
+     * Search and fetch for a partner with a given user or partner id.
+     * @param {Object} param0
+     * @param {number} param0.userId
+     * @param {number} param0.partnerId
+     * @returns {Promise<import("models").Thread | undefined>}
+     */
+    async getChat({ userId, partnerId }) {
+        const partner = await this.getPartner({ userId, partnerId });
+        let chat = partner?.searchChat();
+        if (!chat || !chat.is_pinned) {
+            chat = await this.joinChat(partnerId || partner?.id);
+        }
+        if (!chat) {
+            this.env.services.notification.add(
+                _t("An unexpected error occurred during the creation of the chat."),
+                { type: "warning" }
+            );
+            return;
+        }
+        return chat;
+    }
+
+    getDiscussSidebarCategoryCounter(categoryId) {
+        return this.DiscussAppCategory.get({ id: categoryId }).threads.reduce((acc, channel) => {
+            if (categoryId === "channels") {
+                return channel.message_needaction_counter > 0 ? acc + 1 : acc;
+            } else {
+                return channel.message_unread_counter > 0 ? acc + 1 : acc;
+            }
+        }, 0);
+    }
+
+    /**
+     * Get the parameters to pass to the message post route.
+     */
+    async getMessagePostParams({
```

Comment:
I think this one should be moved on thread model rather than having thread as param.

Reply:
Probably, `thread` was not 1st param so it gave wrong impression it could be optional, hence why I avoided putting in `Thread` and instead put it in `Store`. At a glance this should be ok to move to `Thread`